test/functional/test-impression.js

```diff
@@ -180,14 +180,15 @@ describe('impression', () => {
     viewer.getParam.withArgs('click').returns('https://www.example.com');
     sandbox.stub(viewer, 'sendMessageAwaitResponse').callsFake(message => {
       if (message == 'getReplaceUrl') {
-        return Promise.resolve('test_location?gclid=1234&foo=bar&example=123');
+        return Promise.resolve({
+          'replaceUrl': 'test_location?gclid=1234&foo=bar&example=123'
```

Comment:
The API will return an absolute URL that looks (roughly) like this:
https://news-potato-com.cdn.ampproject.org/v/s/news.potato.com/?gclid=xx&amp_js_v=1

Not sure if your tests need to be that specific but I figured I'd mention here just to be safe :)

Reply:
Good idea! I changed the message to what you provided : )